builtin/gc.c

```diff
@@ -1451,9 +1473,11 @@ static int maintenance_run(int argc, const char **argv, const char *prefix)
 	 * the gvfs.sharedcache config option to redirect the
 	 * maintenance to that location.
 	 */
-	if (!git_config_get_value("gvfs.sharedcache", &object_dir) &&
-	    object_dir)
+	if (!git_config_get_value("gvfs.sharedcache", &tmp_obj_dir) &&
```

Comment:
yeah, i the the problem here is that git_config_get_value is returning a pointer inside of the buffer used to parse/load the config data (such that the next call to this routine might overwrite the buffer) and we're holding it too long here.